qa/tasks/cephfs/test_client_recovery.py

```diff
@@ -567,3 +567,34 @@ def test_abort_conn(self):
         self.assert_session_state(gid, "open")
         time.sleep(session_timeout * 1.5)  # Long enough for MDS to consider session stale
         self.assert_session_state(gid, "stale")
+
+    def test_dont_mark_unresponsive_client_stale(self):
+        """
+        Test that an unresponsive client holding caps not required by any
+        other client is evicted directly, without being marked as stale even
+        after session_timeout, at session_autoclose.
+        """
+        # XXX: To conduct this test we need at least two clients since a
+        # single client is never evcited by MDS.
+        SESSION_TIMEOUT = 30
+        SESSION_AUTOCLOSE = 50
+        mount_a_gid = self.mount_a.get_global_id()
+        mount_a_pid = self.mount_a.get_client_pid()
```

Comment:
````
2019-06-15T22:15:55.528 INFO:tasks.cephfs_test_runner:======================================================================
2019-06-15T22:15:55.528 INFO:tasks.cephfs_test_runner:ERROR: test_dont_mark_unresponsive_client_stale (tasks.cephfs.test_client_recovery.TestClientRecovery)
2019-06-15T22:15:55.528 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-06-15T22:15:55.528 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-06-15T22:15:55.529 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20190614.222049/qa/tasks/cephfs/test_client_recovery.py", line 582, in test_dont_mark_unresponsive_client_stale
2019-06-15T22:15:55.529 INFO:tasks.cephfs_test_runner:    mount_a_pid = self.mount_a.get_client_pid()
2019-06-15T22:15:55.529 INFO:tasks.cephfs_test_runner:AttributeError: 'KernelMount' object has no attribute 'get_client_pid'
````

From: /ceph/teuthology-archive/pdonnell-2019-06-15_02:00:55-kcephfs-wip-pdonnell-testing-20190614.222049-distro-basic-smithi/4035509/teuthology.log

Suggest you skip testing the kernel client. grep for "skipTest" elsewhere in the qa suite on how to do that.

Reply:
Made the change. PTAL.